homeassistant/components/sensor/bom.py

```diff
@@ -211,17 +213,41 @@ def get_reading(self, condition):
         for the latest value that is not `-`.
 
         Iterators are used in this method to avoid iterating needlessly
-        iterating through the entire BOM provided dataset.
+        through the entire BOM provided dataset.
         """
         condition_readings = (entry[condition] for entry in self._data)
         return next((x for x in condition_readings if x != '-'), None)
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
-        """Get the latest data from BOM."""
+        """Get the latest data from BOM.
+
+        Check if it is > 35 minutes since the last reading came in.
+        The readings are timestamped on the hour/half hour and the
+        data generally refreshes 35 minutes after that.
+        You don't want to just throttle for 35 mins as that doesn't take
+        into account how far you currently are from the last reading time.
+        """
+        if self._lastupdate != 0 and \
+            ((datetime.datetime.now() - self._lastupdate) <
+             datetime.timedelta(minutes=35)):
+            _LOGGER.debug(
+                "BOM was updated %s minutes ago, skipping update as"
+                " < 35 minutes, Now: %s, LastUpdate: %s",
+                (datetime.datetime.now() - self._lastupdate),
+                datetime.datetime.now(), self._lastupdate)
+            return self._lastupdate
+
         try:
             result = requests.get(self._build_url(), timeout=10).json()
             self._data = result['observations']['data']
+
+            # set lastupdate using self._data[0] as the first element in the
+            # array is the latest date in the json
+            self._lastupdate = datetime.datetime.strptime(
+                str(self._data[0]['local_date_time_full']), '%Y%m%d%H%M%S')
```

Comment:
Is there a reason that there needs to be a `str(` cast here?

Reply:
I noticed this exists here, but perhaps it's something that can be removed?


https://github.com/home-assistant/home-assistant/blob/e0c8e1dca554229bfbe879707caba0b3b349c7be/homeassistant/components/sensor/bom.py#L164